apps/dashboard/app/apps/user_preferences.rb

```diff
@@ -0,0 +1,48 @@
+class UserPreferences
+    require 'yaml'
+  
+    def initialize()
+        FileUtils.mkdir_p("~/.config/ondemand/") unless File.exists?("~/.config/ondemand/")
+        @preferences ||= begin
+            if preferences_file.file? && preferences_file.readable?
+                YAML.safe_load(preferences_file.read)
+            else
+                {}
+            end
+        rescue => e
+            puts "#{e.message}"
+            @preferences = {}
+        end
+    end
+
+    attr_accessor :preferences
+  
+    def preferences_file
+        @preferences_file ||= begin
+            file = ENV['OOD_PREFERENCES_FILE'] || "#{User.new.dir}/.config/ondemand/preferences.yml"
+            Pathname.new(file.to_s).expand_path
+        end
+    end
+  
+    def method_missing(method_name, *args, &block)
+        method_string = method_name.to_s
+        if @preferences.key?(method_string)
+            @preferences.dig(method_string, *args)
+        else
+            super
+        end
+    end
```

Comment:
I can't tell what this does off hand, especially with `args` (I'm not sure that we should allow args).  I also thing it should just return `nil` if it can't find what it's looking for (dig probably does this already).

Some test cases around this would be helpful.

Reply:
The args are provided for dig 
ex.
`pref.dashboard('color')`

would be interpreted as 
`pref.dig('dashboard', 'color')`

Reply:
Cool. The logic is sound then, and that's fine. Not sure about `super` or casting the method to a string, hashes I've found typically symbolize keys by default. In any case, test cases will show how it works, or should work.

Seems like this hash, `@preferences`, should just deep symbolize keys just so we standardize on that 1 way of referencing them.

Reply:
When I say I'm not sure about `super` and that whole if/else block, looks like `dig` is smart enough to just return nils if it's not found and I think that's just fine.

```ruby
2.7.1 :001 > {}.dig(:foo)
 => nil 
2.7.1 :002 > {}.dig(:foo, :bar)
 => nil 
2.7.1 :003 > {}.dig(:foo, :bar, :third_thing)
 => nil 
```